src/backend/utils/fmgr/fmgr.c

```diff
@@ -821,14 +823,10 @@ fmgr_security_definer(PG_FUNCTION_ARGS)
 		}
 
 		ReleaseSysCache(tuple);
-		sql_dialect_value_old = GetConfigOption("babelfishpg_tsql.sql_dialect", true, true);
-		set_config_option("babelfishpg_tsql.sql_dialect", sql_dialect_value,
-						  (superuser() ? PGC_SUSET : PGC_USERSET),
-						  PGC_S_SESSION,
-						  GUC_ACTION_SAVE,
-						  true,
-						  0,
-						  false);
+		sql_dialect_value_old = sql_dialect;
```

Comment:
do we have some simple test case which shows benefit of this change? Ideally, it should be a function/proc where cross dialect calls are done in each statement and the statements are executed in a loop

Reply:
Addressed in PR description